### PR TITLE
Add console events and new events for processes

### PR DESCRIPTION
**Pull request by joelwurtz**

### CODE REVIEW

File: doc/going-further/extending-castor/events.md [hunk 1]
```diff
@@ -39,3 +39,15 @@ Here is the built-in events triggered by Castor:
 
 * `Castor\Event\AfterExecuteTaskEvent`: This event is triggered after executing
   a task. It provides access to the `TaskCommand` instance.
+
+* `Castor\Event\ProcessStartEvent`: This event is triggered after a process has
+  been started. It provides access to the `Process` instance and the current `Command`.
+
+* `Castor\Event\ProcessTerminateEvent`: This event is triggered after a process has
+  been terminated. It provides access to the `Process` instance and the current `Command`.
```
Comment (pyrech) [inline doc/going-further/extending-castor/events.md:47]: I would also precise that process are created through the `run()` function (i think not everyone will make the link)

File: doc/going-further/extending-castor/events.md [hunk 2]
```diff
@@ -39,3 +39,15 @@ Here is the built-in events triggered by Castor:
...
+
+## Console events
```
Comment (pyrech) [inline doc/going-further/extending-castor/events.md:49]: ♥️

**Review by pyrech — APPROVED** [review]

File: src/Console/Application.php [hunk 3]
```diff
@@ -90,6 +90,7 @@ public function __construct(
         ]);
 
         $this->setCatchErrors(true);
+        $this->setDispatcher($eventDispatcher);
```
Comment (lyrixx) [inline src/Console/Application.php:93]: Could you move this to the ApplicationFactory?

**Review by lyrixx — COMMENTED** [review]
Nice!

**Review by lyrixx — APPROVED** [review]
👍🏼 
Could you update the CHANGELOG?

**Review by pyrech — APPROVED** [review]

**Comment by lyrixx** [comment]
thanks

**Pull request merged**